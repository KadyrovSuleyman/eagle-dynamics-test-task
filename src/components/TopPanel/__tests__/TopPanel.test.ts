import Vuex, { Store } from 'vuex';
import { createLocalVue, mount } from '@vue/test-utils';
import TopPanelVue from '../TopPanel.vue';

jest.mock('../state.ts');

const localVue = createLocalVue();
localVue.use(Vuex);

let store: Store<unknown>;
beforeEach(() => {
  store = new Vuex.Store({
    state: {
      online: 10,
      total: 100,
    },
    mutations: {
      change: (state) => {
        state.online = 7;
        state.total = 50;
      },
    },
  });
});

it('renders', () => {
  const wrapper = mount(TopPanelVue, { localVue, store });
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs outer store', async () => {
  const wrapper = mount(TopPanelVue, { localVue, store });
  expect(wrapper.exists()).toBeTruthy();
  expect(wrapper.find('.counter-online').text()).toBe('10');
  expect(wrapper.find('.counter-total').text()).toBe('100');

  store.commit('change');
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.counter-online').text()).toBe('7');
  expect(wrapper.find('.counter-total').text()).toBe('50');
});
