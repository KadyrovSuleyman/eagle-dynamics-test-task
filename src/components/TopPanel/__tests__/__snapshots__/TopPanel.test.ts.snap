// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div class="app-topPanel"><span class="topPanel-header">
    Список игроков
  </span>
  <div class="topPanel-counter"><span class="counter-text">
      Подключено:
    </span> <span class="counter-online">
      10
    </span> <span class="counter-separator">
      /
    </span> <span class="counter-total">
      100
    </span> <span class="counter-explanation">
      игроков
    </span></div>
</div>
`;
